Is there any way to solve this problem?

This code worked fine when using v5.
After using v6, the following error occurs.
Can anyone let me know if anyone has encountered this error or solved it?

- My Code
```
try {
      const expirationTime = Math.round(Date.now() / 1000 + 60 * 60 * 24 * 90);
      console.log('getSeaport()', await getSeaport());
      const listing = await getSeaport().createSellOrder({
        asset: {
          tokenId,
          tokenAddress: REACT_APP_ETHEREUM_MINTING_CONTRACT_ADDRESS,
        },
        accountAddress: account,
        startAmount: price,
        endAmount: price,
        expirationTime,
      });
      setTxHash(listing);
      setTxLoading(false);
    } catch (e) {
      console.error(e);
      toastError('Listing failed', 'Please try again');
      resetTxHash();
      setTxLoading(false);
    }
```

- Error
**vendors~main.chunk.js:111276 Error: processing response error (body="{\"jsonrpc\":\"2.0\",\"id\":49,\"error\":{\"code\":-32601,\"message\":\"The method eth_signTypedData_v4 does not exist/is not available\"}}", error={"code":-32601}, requestBody="{\"method\":\"eth_signTypedData_v4\",\"params\":[\"0xc0751b9054be33caae11d415281cac9591ad3ef8\",\"{\\\"types\\\":{\\\"OrderComponents\\\":[{\\\"name\\\":\\\"offerer\\\",\\\"type\\\":\\\"address\\\"},{\\\"name\\\":\\\"zone\\\",\\\"type\\\":\\\"address\\\"},{\\\"name\\\":\\\"offer\\\",\\\"type\\\":\\\"OfferItem[]\\\"},{\\\"name\\\":\\\"consideration\\\",\\\"type\\\":\\\"ConsiderationItem[]\\\"},{\\\"name\\\":\\\"orderType\\\",\\\"type\\\":\\\"uint8\\\"},{\\\"name\\\":\\\"startTime\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"endTime\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"zoneHash\\\",\\\"type\\\":\\\"bytes32\\\"},{\\\"name\\\":\\\"salt\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"conduitKey\\\",\\\"type\\\":\\\"bytes32\\\"},{\\\"name\\\":\\\"counter\\\",\\\"type\\\":\\\"uint256\\\"}],\\\"OfferItem\\\":[{\\\"name\\\":\\\"itemType\\\",\\\"type\\\":\\\"uint8\\\"},{\\\"name\\\":\\\"token\\\",\\\"type\\\":\\\"address\\\"},{\\\"name\\\":\\\"identifierOrCriteria\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"startAmount\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"endAmount\\\",\\\"type\\\":\\\"uint256\\\"}],\\\"ConsiderationItem\\\":[{\\\"name\\\":\\\"itemType\\\",\\\"type\\\":\\\"uint8\\\"},{\\\"name\\\":\\\"token\\\",\\\"type\\\":\\\"address\\\"},{\\\"name\\\":\\\"identifierOrCriteria\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"startAmount\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"endAmount\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"recipient\\\",\\\"type\\\":\\\"address\\\"}],\\\"EIP712Domain\\\":[{\\\"name\\\":\\\"name\\\",\\\"type\\\":\\\"string\\\"},{\\\"name\\\":\\\"version\\\",\\\"type\\\":\\\"string\\\"},{\\\"name\\\":\\\"chainId\\\",\\\"type\\\":\\\"uint256\\\"},{\\\"name\\\":\\\"verifyingContract\\\",\\\"type\\\":\\\"address\\\"}]},\\\"domain\\\":{\\\"name\\\":\\\"Seaport\\\",\\\"version\\\":\\\"1.5\\\",\\\"chainId\\\":\\\"5\\\",\\\"verifyingContract\\\":\\\"0x00000000000000adc04c56bf30ac9d3c0aaf14dc\\\"},\\\"primaryType\\\":\\\"OrderComponents\\\",\\\"message\\\":{\\\"offerer\\\":\\\"0xc0751b9054be33caae11d415281cac9591ad3ef8\\\",\\\"zone\\\":\\\"0x0000000000000000000000000000000000000000\\\",\\\"offer\\\":[{\\\"itemType\\\":\\\"2\\\",\\\"token\\\":\\\"0x97911390baef999af13ea7fa0bf60dbaa1362ec7\\\",\\\"identifierOrCriteria\\\":\\\"45\\\",\\\"startAmount\\\":\\\"1\\\",\\\"endAmount\\\":\\\"1\\\"}],\\\"consideration\\\":[{\\\"itemType\\\":\\\"0\\\",\\\"token\\\":\\\"0x0000000000000000000000000000000000000000\\\",\\\"identifierOrCriteria\\\":\\\"0\\\",\\\"startAmount\\\":\\\"975000000000000000\\\",\\\"endAmount\\\":\\\"975000000000000000\\\",\\\"recipient\\\":\\\"0xc0751b9054be33caae11d415281cac9591ad3ef8\\\"},{\\\"itemType\\\":\\\"0\\\",\\\"token\\\":\\\"0x0000000000000000000000000000000000000000\\\",\\\"identifierOrCriteria\\\":\\\"0\\\",\\\"startAmount\\\":\\\"25000000000000000\\\",\\\"endAmount\\\":\\\"25000000000000000\\\",\\\"recipient\\\":\\\"0x0000a26b00c1f0df003000390027140000faa719\\\"}],\\\"orderType\\\":\\\"1\\\",\\\"startTime\\\":\\\"1689568299\\\",\\\"endTime\\\":\\\"1697344298\\\",\\\"zoneHash\\\":\\\"0x0000000000000000000000000000000000000000000000000000000000000000\\\",\\\"salt\\\":\\\"0\\\",\\\"conduitKey\\\":\\\"0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000\\\",\\\"counter\\\":\\\"0\\\"}}\"],\"id\":49,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://goerli.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161", code=SERVER_ERROR, version=web/5.7.1)
    at Logger.makeError (vendors~main.chunk.js:12570:19)
    at Logger.throwError (vendors~main.chunk.js:12581:18)
    at _callee$ (vendors~main.chunk.js:28748:20)
    at tryCatch (vendors~main.chunk.js:2255:17)
    at Generator.<anonymous> (vendors~main.chunk.js:2336:22)
    at Generator.next (vendors~main.chunk.js:2280:21)
    at fulfilled (vendors~main.chunk.js:28410:24)**

--- a/src/sdk.ts
+++ b/src/sdk.ts
@@ -156,6 +156,8 @@ export class OpenSeaSDK {
     amountInEth: BigNumberish;
     accountAddress: string;
   }) {
+    await this._checkAccountIsAvailable(accountAddress);
+
     const value = parseEther(FixedNumber.from(amountInEth).toString());
 
     this._dispatch(EventType.WrapEth, { accountAddress, amount: value });
@@ -193,6 +195,8 @@ export class OpenSeaSDK {
     amountInEth: BigNumberish;
     accountAddress: string;
   }) {
+    await this._checkAccountIsAvailable(accountAddress);
+
     const amount = parseEther(FixedNumber.from(amountInEth).toString());
 
     this._dispatch(EventType.UnwrapWeth, { accountAddress, amount });
@@ -350,6 +354,8 @@ export class OpenSeaSDK {
     expirationTime?: BigNumberish;
     paymentTokenAddress?: string;
   }): Promise<OrderV2> {
+    await this._checkAccountIsAvailable(accountAddress);
+
     if (!asset.tokenId) {
       throw new Error("Asset must have a tokenId");
     }
@@ -454,6 +460,8 @@ export class OpenSeaSDK {
     paymentTokenAddress?: string;
     buyerAddress?: string;
   }): Promise<OrderV2> {
+    await this._checkAccountIsAvailable(accountAddress);
+
     if (!asset.tokenId) {
       throw new Error("Asset must have a tokenId");
     }
@@ -544,6 +552,8 @@ export class OpenSeaSDK {
     expirationTime?: number | string;
     paymentTokenAddress: string;
   }): Promise<PostOfferResponse | null> {
+    await this._checkAccountIsAvailable(accountAddress);
+
     const collection = await this.api.getCollection(collectionSlug);
     const buildOfferResult = await this.api.buildOffer(
       accountAddress,
@@ -614,10 +624,6 @@ export class OpenSeaSDK {
     accountAddress: string;
     domain?: string;
   }): Promise<string> {
-    if (!isValidProtocol(order.protocolAddress)) {
-      throw new Error("Unsupported protocol");
-    }
-
     if (!order.taker?.address) {
       throw new Error(
         "Order is not a private listing must have a taker address",
@@ -669,6 +675,8 @@ export class OpenSeaSDK {
     recipientAddress?: string;
     domain?: string;
   }): Promise<string> {
+    await this._checkAccountIsAvailable(accountAddress);
+
     if (!isValidProtocol(order.protocolAddress)) {
       throw new Error("Unsupported protocol");
     }
@@ -752,6 +760,8 @@ export class OpenSeaSDK {
     accountAddress: string;
     domain?: string;
   }) {
+    await this._checkAccountIsAvailable(accountAddress);
+
     if (!isValidProtocol(order.protocolAddress)) {
       throw new Error("Unsupported protocol");
     }
@@ -915,6 +925,8 @@ export class OpenSeaSDK {
    * @returns Transaction hash of the approval transaction
    */
   public async approveOrder(order: OrderV2, domain?: string) {
+    await this._checkAccountIsAvailable(order.maker.address);
+
     if (!isValidProtocol(order.protocolAddress)) {
       throw new Error("Unsupported protocol");
     }
@@ -1042,6 +1054,28 @@ export class OpenSeaSDK {
 
   private _dispatch(event: EventType, data: EventData) {
     this._emitter.emit(event, data);
+  }
+
+  /**
+   * Throws an error if an account is not available through the provider.
+   *
+   * @param accountAddress The account address to check is available.
+   */
+  private async _checkAccountIsAvailable(accountAddress: string) {
+    const accountAddressChecksummed = ethers.utils.getAddress(accountAddress);
+    if (
+      (this._signerOrProvider as Wallet).address !==
+        accountAddressChecksummed &&
+      !(
+        await (
+          this._signerOrProvider as providers.JsonRpcProvider
+        ).listAccounts()
+      ).includes(accountAddressChecksummed)
+    ) {
+      throw new Error(
+        `Specified accountAddress is not available through wallet or provider: ${accountAddressChecksummed}`,
+      );
+    }
   }
 
   private async _confirmTransaction(

--- a/test/sdk/misc.spec.ts
+++ b/test/sdk/misc.spec.ts
@@ -1,4 +1,5 @@
-import { assert } from "chai";
+import { assert, expect } from "chai";
+import { ethers } from "ethers";
 import { suite, test } from "mocha";
 import {
   SHARED_STOREFRONT_LAZY_MINT_ADAPTER_CROSS_CHAIN_ADDRESS,
@@ -71,5 +72,82 @@ suite("SDK: misc", () => {
       ),
       SHARED_STOREFRONT_LAZY_MINT_ADAPTER_CROSS_CHAIN_ADDRESS,
     );
+  });
+
+  test("Should throw an error when using methods that need a provider or wallet with the accountAddress", async () => {
+    const wallet = ethers.Wallet.createRandom();
+    const accountAddress = wallet.address;
+    const expectedErrorMessage = `Specified accountAddress is not available through wallet or provider: ${accountAddress}`;
+
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    try {
+      await client.wrapEth({ amountInEth: "0.1", accountAddress });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    try {
+      await client.unwrapWeth({ amountInEth: "0.1", accountAddress });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    const asset = {} as any;
+
+    try {
+      await client.createBuyOrder({ asset, startAmount: 1, accountAddress });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    try {
+      await client.createSellOrder({ asset, startAmount: 1, accountAddress });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    try {
+      await client.createCollectionOffer({
+        collectionSlug: "",
+        amount: 1,
+        quantity: 1,
+        paymentTokenAddress: "",
+        accountAddress,
+      });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    const order = {} as any;
+
+    try {
+      await client.fulfillOrder({ order, accountAddress });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    try {
+      await client.cancelOrder({ order, accountAddress });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+
+    try {
+      await client.approveOrder({
+        ...order,
+        maker: { address: accountAddress },
+      });
+      throw new Error("should have thrown");
+    } catch (e: any) {
+      expect(e.message).to.equal(expectedErrorMessage);
+    }
+    /* eslint-enable @typescript-eslint/no-explicit-any */
   });
 });

